similarities/models.py

```diff
@@ -68,25 +71,41 @@ class Meta:
         )
 
     def save(self, *args, **kwargs):
-        similarities_to_update = []
+        other_artists_to_update = []
         if self.pk:
             prev_other_artist = type(self).objects.get(pk=self.pk).other_artist
         else:
             prev_other_artist = None
         super().save(*args, **kwargs)
         if self.other_artist:
-            cumulative_similarity, _ = Similarity.objects.get_or_create(
-                other_artist=self.other_artist,
-                cc_artist=self.cc_artist,
-            )
-            similarities_to_update.append(cumulative_similarity)
+            other_artists_to_update.append(self.other_artist)
         if prev_other_artist and self.other_artist != prev_other_artist:
-            old_cumulative_similarity, _ = Similarity.objects.get_or_create(
-                other_artist=prev_other_artist,
+            other_artists_to_update.append(prev_other_artist)
+        for other in other_artists_to_update:
+            Similarity.objects.update_or_create_by_artists(
+                other_artist=other,
                 cc_artist=self.cc_artist,
             )
-            similarities_to_update.append(old_cumulative_similarity)
-        update_similarities(similarities_to_update)
+
+
+class SimilarityManager(models.Manager):
+
+    def update_or_create_by_artists(self, other_artist, cc_artist):
+        low_track_links = (cc_artist.links
+                           .filter(num_tracks__lt=MIN_TRACKS_TO_HAVE_WEIGHT)
+                           .exclude(num_tracks__gte=MIN_TRACKS_TO_HAVE_WEIGHT))
+        if low_track_links.exists():
```

Comment:
If the artist has 0 tracks on Jamendo but 20 on FreeMusicArchive they won't show up, right?  If so, this seems like a bug.